GoogleForCreators/web-stories-wp
# Editor: Add Semantic Headings panel to Style tab

## Context


This PR allows a user to select one or multiple text elements within the canvas and either view or change the semantic elements used to render text.

## Summary


This PR adds a new experiment. When this experiment is enabled, a user can change the semantic heading of an element of text within the Editor. This does NOT alter the size or style of the text, only the semantic element within the output HTML (such as an `h1` or `h3`, etc).

## Relevant Technical Choices



## To-do



## User-facing changes



<img width="352" alt="Screen Shot 2022-07-08 at 1 21 23 PM" src="https://user-images.githubusercontent.com/7110244/178040366-d141a8e2-7cb1-47a8-a5dc-aedbd4c19745.png">

https://user-images.githubusercontent.com/841956/179558369-af622a63-f470-4ca7-9cb7-3e8803f45b5f.mov

## Testing Instructions






This PR can be tested by following these steps:

1. Enable the `Semantic Headings` experiment in Stories > Experiments.
2. Navigate to a Story within the Editor.
3. Add multiple text elements with variying font sizes.
4. You should be able to see in the Style section of the sidebar, underneath a new Accessibility tab, the type of semantic heading.
5. You should be able to change the semantic heading, and publish your changes.
6. After this is done, the element within the dev tools should reflect the semantic heading change.
7. You should also be able to do this for multiple text elements at once (if the text elements have mismatching types when initially selected, i.e. a Heading 1 and a Heading 2, you should see "Mixed" as the current value in the dropdown).

Things to confirm:

* If multiple items that are Automatic are selected, it should show 'Automatic (Mixed)'
* If two items are selected, and one was set automatic and the other was not, it should be indicated that it is Mixed.
* When there are two large elements with equal font sizes are set to "Automatic", and you move one of them higher up on the canvas, it should become the h1, and if you move it down, it will become the h2 (the dropdown should update accordingly)

## Reviews

### Does this PR have a security-related impact?



No

### Does this PR change what data or activity we track or use?



No

### Does this PR have a legal-related impact?



No

## Checklist




---



Fixes #11788
Fixes #11916
